### PR TITLE
Add basic liveness and readiness probe

### PR DESCRIPTION
**Pull request by stuggi**
Might require tuning.

### CODE REVIEW

File: pkg/keystone/deployment.go [hunk 1]
```diff
@@ -37,11 +38,43 @@ func Deployment(
...
+
+		//
+		// https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
+		//
+		livenessProbe.HTTPGet = &corev1.HTTPGetAction{
+			Port: intstr.IntOrString{Type: intstr.Int, IntVal: int32(KeystonePublicPort)},
+		}
+		readinessProbe.HTTPGet = &corev1.HTTPGetAction{
+			Port: intstr.IntOrString{Type: intstr.Int, IntVal: int32(KeystonePublicPort)},
+		}
```
Comment (abays) [inline pkg/keystone/deployment.go:77]: Is the default `/` path okay for these?  I'm guessing it is

Comment (stuggi) [reply pkg/keystone/deployment.go:77]: yes, see  [1] , which redirects to the admin api [2] and / is to "Discover API version information". The check could probably improved, but should work for an initial start.

[1] https://docs.openstack.org/keystone/pike/api_curl_examples.html#get
[2] https://docs.openstack.org/keystone/pike/api_curl_examples.html#id1

**Review by abays — APPROVED** [review]
/lgtm

**Comment by stuggi** [comment]
had to rebase

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**